.github/workflows/main.yml

```diff
@@ -2,40 +2,287 @@ name: CI
+        timeout-minutes: 3
+      - name: Preconnect to cluster peers
+        run: |
+          echo '-> preconnect to cluster peers'
+          ipfs-cluster-ctl --enc=json \
+            --host "/dnsaddr/ipfs-websites.collab.ipfscluster.io" \
+            --basic-auth '${{ secrets.CLUSTER_USER }}:${{ secrets.CLUSTER_PASSWORD }}' \
+            peers ls > cluster-peers-ls
+          for maddr in $(jq -r '.[].ipfs.addresses[]?' cluster-peers-ls); do
+            ipfs swarm connect "$maddr" || continue
+          done
+          echo '-> manual connect to cluster.ipfs.io'
+          ipfs swarm connect /dnsaddr/cluster.ipfs.io
+          echo '-> list swarm peers'
+          ipfs swarm peers
+        timeout-minutes: 3
+      - run: ./dockerized make publish
+      - run: git status
+      - name: Read CID of updated DAG
+        id: cid-reader
+        run: echo "::set-output name=CID::$(tail -1 ./versions)"
+      - name: Pin new website to ipfs-websites.collab.ipfscluster.io
+        run: |
+          echo 'pin to cluster'
+          ipfs-cluster-ctl --enc=json \
+            --host "/dnsaddr/ipfs-websites.collab.ipfscluster.io" \
+            --basic-auth '${{ secrets.CLUSTER_USER }}:${{ secrets.CLUSTER_PASSWORD }}' \
+            pin add \
+            --pin-name="https://github.com/ipfs/distributions/commits/${{ github.sha }}" \
+            --no-status \
+            "${{ steps.cid-reader.outputs.CID }}"
+          while true; do
+            ipfs-cluster-ctl --enc=json \
+              --host "/dnsaddr/ipfs-websites.collab.ipfscluster.io" \
+              --basic-auth '${{ secrets.CLUSTER_USER }}:${{ secrets.CLUSTER_PASSWORD }}' \
+              status "${{ steps.cid-reader.outputs.CID }}" | tee cluster-pin-status
+            if [[ $(jq '.peer_map[].status' cluster-pin-status | grep '"pinned"' | wc -l) -ge 2 ]]; then
+              echo "Got 2 pin confirmations, finishing the workflow"
+              break
+            else
+              echo "(sleeping for 15 seconds)"
+              sleep 15
+            fi
+          done
+        timeout-minutes: 60
+      - name: Update PR status with preview link
+        run: |
+          PREVIEW_URL="https://dweb.link/ipfs/${{ steps.cid-reader.outputs.CID }}"
+          API_PARAMS=$(jq --monochrome-output --null-input \
+            --arg state "success" \
+            --arg target_url "$PREVIEW_URL" \
+            --arg description "Preview on IPFS" \
+            --arg context "IPFS" \
+            '{ state: $state, target_url: $target_url, description: $description, context: $context }' )
+          curl --output /dev/null --silent --show-error \
+            -X POST -H 'Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' -H 'Content-Type: application/json' \
+            --data "$API_PARAMS" 'https://api.github.com/repos/ipfs/distributions/statuses/${{ github.sha }}'
+          echo "Pinned to IPFS - $PREVIEW_URL"
```

Comment:
This mechanism for adding a status result to a PRs checks list stopped working for me in ipfs-dns-deploy. I tried to fix it, but could never get the status to appear again on the PR... what did you change to make it work here!?

Reply:
@olizilla Hm.. not sure, could be one of below:
* copied it from https://github.com/ipfs-shipyard/ipfs-github-action/blob/182f36dcd28a0df011ce1ee9b004a1ca7fa7c720/entrypoint.sh#L11-L32 and tweaked quotes and renamed some fields.
* stopped installing `jq`  via snap and use the version shipped with `ubuntu-latest`. 
* use `GITHUB_TOKEN` secret directly, and not via env
  * I would guess this is most likely the cause. Github hardened some things around their tokens this year, [enabling orgs to tweak defaults or limit the scope](https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/) (maybe PL changed something?). I've read that some github actions stopped working with regular `GITHUB_TOKEN` and require creation of a [a PAT (Personal Access Token)](https://docs.github.com/en/enterprise-server@3.1/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token) (I think only official one under `actions/*`  continue to work "old way", but also did not dig into this dep enough to understand what exactly changed and why)